src/diffusers/pipelines/auto_pipeline.py

```diff
@@ -953,7 +953,11 @@ def from_pretrained(cls, pretrained_model_or_path, **kwargs):
         if "enable_pag" in kwargs:
             enable_pag = kwargs.pop("enable_pag")
             if enable_pag:
-                orig_class_name = config["_class_name"].replace("Pipeline", "PAGPipeline")
+                orig_class_name = (
```

Comment:
if we do this,  the `orig_class_name` for `StableDiffusionXLPipeline` would be `StableDiffusionXLPAGInpaintPipeline` too - in this case will get the correct result regardless but logic is incorrect

the `orig_class_name` for `StableDiffusionXLPipeline` should be `StableDiffusionXLPAGPipeline` and then we map it to the corresponding to the inpainting class through `_get_task_class(...)` and get `StableDiffusionXLPAGInpaintPipeline`

we can maybe do something like 

```python
to_replace = "InpaintPipeline" if "Inpaint` in config["_class_name"] else "Pipeline"
orig_class_name = config["_class_name"].replace(to_replace, "PAG" + to_replace)
```